File: backend/app/schemas/filial.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime

class FilialBase(BaseModel):
    nome: str
    cnpj: Optional[str] = None
    is_matriz: bool = False
    ativo: bool = True

class FilialCriar(FilialBase):
    pass

class FilialSchema(FilialBase):
    id: int
    criado_em: datetime
    atualizado_em: datetime
    rowversion: int

    class Config:
        from_attributes = True

File: frontend/src/types/filial.ts
export interface Filial {
  id: number;
  nome: string;
  cnpj?: string | null;
  url_api?: string | null;
  is_matriz: boolean;
  ativo: boolean;
  criado_em: string;
  atualizado_em: string;
  rowversion: number;
}

export interface FilialCriar {
  nome: string;
  cnpj?: string | null;
  url_api?: string | null;
  is_matriz: boolean;
  ativo: boolean;
}